imports/plugins/included/payments-stripe/server/methods/stripeapi.js

```diff
@@ -43,118 +43,80 @@ const expectedErrors = [
+  const stripeKey = apiKey || StripeApi.methods.getApiKey();
+  const stripe = require("stripe")(stripeKey);
+  try {
+    const chargePromise = stripe.charges.create(chargeObj);
+    const promiseResult = Promise.await(chargePromise);
+    return promiseResult;
+  } catch (e) {
+    // Handle "expected" errors differently
+    if (e.rawType === "card_error" && _.includes(expectedErrors, e.code)) {
+      Logger.debug("Error from Stripe is expected, not throwing");
+      const normalizedError = {
+        details: e.message
       };
-      return { error: sanitisedError, result: null };
+      return { error: normalizedError, result: null };
     }
-  }
-});
+    Logger.error("Received unexpected error type: " + e.rawType);
+    Logger.error(e);
 
-StripeApi.methods.captureCharge = new ValidatedMethod({
-  name: "StripeApi.methods.captureCharge",
-  validate: new SimpleSchema({
-    transactionId: { type: String },
-    captureDetails: { type: captureDetailsSchema },
-    apiKey: { type: String, optional: true }
-  }).validator(),
-  run({ transactionId, captureDetails, apiKey })  {
-    let stripe;
-    if (!apiKey) {
-      const dynamicApiKey = StripeApi.methods.getApiKey.call();
-      stripe = require("stripe")(dynamicApiKey);
-    } else {
-      stripe = require("stripe")(apiKey);
-    }
-    const capturePromise = stripe.charges.capture(transactionId, captureDetails);
-    const captureResults = Promise.await(capturePromise);
-    return captureResults;
+    // send raw error to server log, but sanitized version to client
+    const sanitisedError = {
+      details: "An unexpected error has occurred"
+    };
+    return { error: sanitisedError, result: null };
   }
-});
+};
 
-StripeApi.methods.createRefund = new ValidatedMethod({
-  name: "StripeApi.methods.createRefund",
-  validate: new SimpleSchema({
-    refundDetails: { type: refundDetailsSchema },
-    apiKey: { type: String, optional: true }
-  }).validator(),
-  run({ refundDetails, apiKey }) {
-    let stripe;
-    if (!apiKey) {
-      const dynamicApiKey = StripeApi.methods.getApiKey.call();
-      stripe = require("stripe")(dynamicApiKey);
-    } else {
-      stripe = require("stripe")(apiKey);
-    }
-    const refundPromise = stripe.refunds.create({ charge: refundDetails.charge, amount: refundDetails.amount });
-    const refundResults = Promise.await(refundPromise);
-    return refundResults;
-  }
-});
 
-StripeApi.methods.listRefunds = new ValidatedMethod({
-  name: "StripeApi.methods.listRefunds",
-  validate: new SimpleSchema({
-    transactionId: { type: String },
-    apiKey: { type: String, optional: true }
-  }).validator(),
-  run({ transactionId, apiKey }) {
-    let stripe;
-    if (!apiKey) {
-      const dynamicApiKey = StripeApi.methods.getApiKey.call();
-      stripe = require("stripe")(dynamicApiKey);
-    } else {
-      stripe = require("stripe")(apiKey);
-    }
+StripeApi.methods.captureCharge = function ({ transactionId, captureDetails, apiKey }) {
```

Comment:
Once all of this "works" for both direct and regular charges, I'd like to see us use a consistent style from method to method. The `stripe/payment/createCharges` method uses the `export const methods` style vs these using the `StripeApi.methods`let's figure out which direction we want to go here and update the other ones to match.

It seems like the rest of the app uses the `export const methods` style.

@aaronjudd or @machikoyasuda any preference on method naming conventions?

Reply:
Yeah, the `StripeApi` was a convention created before we had import/exports. I think we can eliminate it and combine `stripeapi.js` and `stripe.js` into one file. Most of this was to allow for testing but I question the value of doing it that way now.